File: fizzbuzz.hs
fizzBuzz :: Integer -> String
fizzBuzz n = if null x
             then show n
             else x
             where x = concat [w | (w, d) <- [("fizz", 3), ("buzz", 5)], mod n d == 0]

main = mapM_ putStrLn [fizzBuzz n | n <- [1..15]]

File: fizzbuzz.m
for i = 1:15
  s = '';
  if mod(i, 3) == 0
    s = strcat(s, 'fizz');
  end
  if mod(i, 5) == 0
    s = strcat(s, 'buzz');
  end
  if isempty(s)
    s = i;
  end
  disp(s);
end
